### PR TITLE
Sdk add key

### PR DESCRIPTION
**Pull request by Sajma27**
# Summary
A description of the changes proposed in the pull request.

Fixes #number

## Checklist
- [ ] Change is small and easy to review (please split big changes into multiple PRs)
- [x] Change is consistent with architecture
- [x] Change is consistent with test architecture
- [x] Change is consistent with naming conventions
- [x] New code is covered with tests
- [x] Tests related to old code are updated
- [x] Documentation is up to date with changes

### CODE REVIEW

File: universal-login-sdk/src/api/SdkSigner.ts [hunk 1]
```diff
@@ -2,19 +2,20 @@ import UniversalLoginSDK from './sdk';
...
   provider: providers.Provider;
 
   constructor(
     private sdk: UniversalLoginSDK,
+    private deployedWallet: DeployedWallet,
     public contractAddress: string,
-    privateKey: string,
   ) {
     super();
     this.provider = sdk.provider;
```
Comment (Marik-D) [inline universal-login-sdk/src/api/SdkSigner.ts]: You can get the provider from `deployedWallet.sdk.provider` and remove the constructor param

File: universal-login-sdk/src/api/SdkSigner.ts [hunk 2]
```diff
@@ -2,19 +2,20 @@ import UniversalLoginSDK from './sdk';
...
+import {DeployedWallet} from './wallet/DeployedWallet';
 
 export class SdkSigner extends ethers.Signer {
   private wallet: Wallet;
   provider: providers.Provider;
 
   constructor(
     private sdk: UniversalLoginSDK,
+    private deployedWallet: DeployedWallet,
     public contractAddress: string,
```
Comment (Marik-D) [inline universal-login-sdk/src/api/SdkSigner.ts]: Also you can get the contract address from deployed wallet

File: docs/source/sdk.rst [hunk 3]
```diff
@@ -164,15 +164,42 @@ denyRequest
...
+    - **name** : string - a name for deployed wallet
+    - **privateKey** : string - a private key to sign a request
+    - **sdk** : object - a UniversalLoginSDK object
+  Returns:
+    DeployedWallet instance
+
+    Example:
+    ::
+
+      import {DeployedWallet} from '../../src';
```
Comment (JustynaBroniszewska) [inline docs/source/sdk.rst]: `import {DeployedWallet} from '@unilogin/sdk';`

File: docs/source/sdk.rst [hunk 4]
```diff
@@ -164,15 +164,89 @@ denyRequest
...
+    - **name** : string - a name for deployed wallet
+    - **privateKey** : string - a private key to sign a request
+    - **sdk** : object - a UniversalLoginSDK object
+  Returns:
+    DeployedWallet instance
+
+    Example:
+    ::
+
+      import {DeployedWallet} from '@unilogin/sdk';
```
Comment (JustynaBroniszewska) [inline docs/source/sdk.rst]: For example: 

```ts
const sdk = new UniversalLoginSDK(....);
const deployedWallet = new DeployedWallet('0x2828282882', 'name.mylogin.eth', '0x1183823828282', sdk);
```

File: docs/source/sdk.rst [hunk 5]
```diff
@@ -164,15 +164,84 @@ denyRequest
...
       const publicKey = await sdk.denyRequest('0xA193E42526F1FEA8C99AF609dcEabf30C1c29fAA', '0xb19Ec9bdC6733Bf0c825FCB6E6Da95518DB80D13');
 
+
+SdkSigner
+^^^^^^^^^
+
+::
+
+  // gasToken should be configured when creating SDK instance in order to use the signer
+  const signer = new SdkSigner(sdk, contractAddress, privateKey);
```
Comment (Marik-D) [inline docs/source/sdk.rst]:
```suggestion
  const signer = new SdkSigner(deployedWallet);
```

File: universal-login-sdk/src/api/SdkSigner.ts [hunk 6]
```diff
@@ -1,23 +1,23 @@
-import UniversalLoginSDK from './sdk';
 import {ethers, Wallet, utils, providers} from 'ethers';
 import {ensureNotFalsy, DEFAULT_GAS_LIMIT} from '@unilogin/commons';
 import {TransactionHashNotFound} from '../core/utils/errors';
+import {DeployedWallet} from './wallet/DeployedWallet';
 
 export class SdkSigner extends ethers.Signer {
   private wallet: Wallet;
   provider: providers.Provider;
+  contractAddress: string;
```
Comment (xitronix) [inline universal-login-sdk/src/api/SdkSigner.ts]: Contract address is present in DeployedWallet

File: universal-login-sdk/src/core/services/WalletService.ts [hunk 7]
```diff
@@ -128,7 +128,8 @@ export class WalletService {
   async recover(name: string, passphrase: string) {
     const contractAddress = await this.sdk.getWalletContractAddress(name);
     const wallet = await this.walletFromPassphrase(name, passphrase);
-    ensure(await this.sdk.keyExist(contractAddress, wallet.address), InvalidPassphrase);
+    const deployedWallet = new DeployedWallet(contractAddress, name, '', this.sdk);
```
Comment (xitronix) [inline universal-login-sdk/src/core/services/WalletService.ts]: this.getDeployedWallet();

File: universal-login-sdk/test/api/sdk.test.ts [hunk 8]
```diff
@@ -63,15 +63,15 @@ describe('INT: SDK', () => {
 
   describe('Execute signed message', () => {
```
Comment (xitronix) [inline universal-login-sdk/test/api/sdk.test.ts]: Move test's which test `DeployedWallet` to deployedWallet.test.ts

File: universal-login-sdk/src/core/services/WalletService.ts (hunk 7 above)
Comment (Sajma27) [reply universal-login-sdk/src/core/services/WalletService.ts]: 'UNIT: WalletService' test doesn’t work with this.getDeployedWallet();

File: universal-login-sdk/test/core/services/WalletService.unit.test.ts [hunk 9]
```diff
@@ -31,6 +31,7 @@ describe('UNIT: WalletService', () => {
       getWalletContractAddress: sinon.stub().withArgs(name).returns(TEST_CONTRACT_ADDRESS),
       keyExist,
```
Comment (xitronix) [inline universal-login-sdk/test/core/services/WalletService.unit.test.ts]: sdk does not have keyExist anymore

File: universal-login-wallet/test/ui/connect.test.tsx [hunk 10]
```diff
@@ -45,7 +46,8 @@ describe('UI: Connection flow', () => {
     appPage.connection().clickConnectWithAnotherDevice();
     await appPage.connection().waitForEmojiView();
     const publicKey = (new Wallet(services.walletService.getConnectingWallet().privateKey)).address;
-    await services.sdk.addKey(contractAddress, publicKey, privateKey, {gasPrice, gasLimit, gasToken});
+    const deployedWallet = new DeployedWallet(contractAddress, '', privateKey, services.sdk);
```
Comment (xitronix) [inline universal-login-wallet/test/ui/connect.test.tsx]:
```suggestion
    const deployedWallet = new DeployedWallet(contractAddress, name, privateKey, services.sdk);
```

**Pull request merged**